linkerd/dns/src/lib.rs

```diff
@@ -112,12 +122,32 @@ impl Resolver {
         &self,
         name: &Name,
     ) -> Pin<Box<dyn Future<Output = Result<net::IpAddr, Error>> + Send + 'static>> {
+        Box::pin(
+            self.resolve_ips(name).map(|rsp| {
+                rsp.map(|result| result.ips.first().expect("would error if empty").clone())
+            }),
+        )
```

Comment:
it's a bummer that we are now double-boxing this future. it might be worth the cost of a little more code duplication to have this method just call `resolver.lookup_ip`, the way that `resolve_ips` does, to avoid having this future be double-boxed.

alternatively, if we are able to change `resolve_ips` to return an `impl Future` as I discussed above, this wouldn't be double boxing...